src/main/drivers/rx_cc2500.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// This file is copied with modifications from OpenSky,
+// see https://github.com/fishpepper/OpenSky
+
+// See also in particular
+// https://github.com/fishpepper/OpenSky/blob/master/arch/stm32f1/hal_cc25xx.c
+// https://github.com/fishpepper/OpenSky/blob/master/cc25xx.h
+// https://github.com/fishpepper/OpenSky/blob/master/cc25xx.c
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <platform.h>
+
+#ifdef USE_RX_CC2500
+
+#include "build/build_config.h"
+
+#include "system.h"
+#include "bus_spi.h"
+#include "io.h"
+#include "rx_spi.h"
+#include "rx_cc2500.h"
+
+static uint8_t cc25xx_current_antenna;
+
+#define CC25XX_RX_HI()       {IOHi(IOGetByTag(IO_TAG(RX_SW_CRX_PIN)));}
```

Comment:
there is `DEFIO_IO(name)` macro to get `IO_t` directly from name (as compile-time constant). It will avoid overhead of  `IOGetByTag` ... 

IO abstraction assumed that `IO_t` for each used pin would be defined and used (so `IOGetByTag` is called only once) ...